examples/mesh-update.rs

```diff
@@ -0,0 +1,75 @@
+extern crate cgmath;
+extern crate three;
+extern crate mint;
+
+use std::f32::consts::PI;
+
+use cgmath::prelude::*;
+
+fn make_tetrahedron_geometry() -> three::Geometry {
+    let vertices = vec![
+        mint::Point3 { x: 0.0, y: 1.0, z: 0.0 },
+        mint::Point3 { x: 0.0, y: 0.0, z: 1.0 },
+        mint::Point3 { x: (2.0 * PI / 3.0).sin(), y: 0.0, z: (2.0 * PI / 3.0).cos() },
+        mint::Point3 { x: (4.0 * PI / 3.0).sin(), y: 0.0, z: (4.0 * PI / 3.0).cos() },
+    ];
+    let faces = vec![
+        [0, 1, 2],
+        [0, 2, 3],
+        [0, 3, 1],
+        [1, 3, 2],
+    ];
+    three::Geometry {
+        faces,
+        base_shape: three::geometry::Shape {
+            vertices,
+            .. three::geometry::Shape::empty()
+        },
+        .. three::Geometry::empty()
+    }
+}
+
+fn main() {
+    let shaders_path = concat!(env!("CARGO_MANIFEST_DIR"), "/data/shaders");
+    let title = "Three-rs Mesh Update Example";
+    let mut win = three::Window::builder(title, &shaders_path).build();
+    let cam = win.factory.perspective_camera(60.0, 1.0 .. 10.0);
+    let mut controls = three::controls::Orbit::builder(&cam)
+        .position([0.0, 2.0, -5.0])
+        .target([0.0, 0.0, 0.0])
+        .build();
+
+    let mut dir_light = win.factory.directional_light(0xFFFFFF, 0.9);
+    dir_light.look_at([15.0, 35.0, 35.0], [0.0, 0.0, 2.0], None);
+    win.scene.add(&dir_light);
+
+    let geometry = make_tetrahedron_geometry();
```

Comment:
Perhaps we should add a `Geometry::tetrahedron` constructor just for this example?

Reply:
Not sure, tetrahedron isn't best shape for everyday usage.
Example became much nicer, thanks!